### PR TITLE
Make `CloudServiceID` configurable in discovery service

### PR DESCRIPTION
**Pull request by oxisto**
This PR makes the cloud service ID for each discovery service configurable. The way it works it that a *discoverer* does not set any service ID and that the *discovery* service sets the corresponding ID in its `StartDiscovery` method.

If no ID is set, it defaults to `discovery.DefaultCloudServiceID`.

### CODE REVIEW

File: voc/voc.go [hunk 1]
```diff
@@ -54,7 +55,8 @@ type ResourceID string
 // Resource file from Ontology currently not used. How do we merge this file with the 'Resource Ontology file'
 type Resource struct {
 	ID ResourceID `json:"id"`
-	// ServiceID contains the ID of the cloud service to which this resource belongs
+	// ServiceID contains the ID of the cloud service to which this resource belongs. When creating new resources in a
+	// discoverer, this can be left out as this will be filled by the discovery service later in the StartDiscovery method.
```
Comment (oxisto) [inline voc/voc.go:59]: This doc needs to be changed, so that `NewResource` should be used.

**Review by anatheka — COMMENTED** [review]
LGTM. Just a minor comment to the CloudServiceID naming and you have an open comment of your own.

File: service/discovery/azure/network.go [hunk 2]
```diff
@@ -37,19 +37,19 @@ import (
...
 type azureNetworkDiscovery struct {
-	azureDiscovery
+	*azureDiscovery
 }
 
 func NewAzureNetworkDiscovery(opts ...DiscoveryOption) discovery.Discoverer {
 	d := &azureNetworkDiscovery{
-		azureDiscovery{
+		&azureDiscovery{
 			discovererComponent: NetworkComponent,
```
Comment (anatheka) [inline service/discovery/azure/network.go:46]:
```suggestion
			discovererComponent: NetworkComponent,
			csi: discovery.DefaultCloudServiceID,
```

File: service/discovery/azure/storage.go [hunk 3]
```diff
@@ -44,19 +44,19 @@ var (
...
 type azureStorageDiscovery struct {
-	azureDiscovery
+	*azureDiscovery
 }
 
 func NewAzureStorageDiscovery(opts ...DiscoveryOption) discovery.Discoverer {
 	d := &azureStorageDiscovery{
-		azureDiscovery: azureDiscovery{
+		&azureDiscovery{
 			discovererComponent: StorageComponent,
```
Comment (anatheka) [inline service/discovery/azure/storage.go:53]:
```suggestion
			discovererComponent: StorageComponent,
			csi: discovery.DefaultCloudServiceID,
```

File: service/discovery/aws/compute.go [hunk 4]
```diff
@@ -132,6 +134,10 @@ func (d computeDiscovery) List() (resources []voc.IsCloudResource, err error) {
 	return
 }
 
+func (d *computeDiscovery) CloudServiceID() string {
```
Comment (anatheka) [inline service/discovery/aws/compute.go:137]: Move that to service/discovery/aws/aws.go?

File: service/discovery/aws/compute.go [hunk 5]
```diff
@@ -47,6 +47,7 @@ type computeDiscovery struct {
 	functionAPI       LambdaAPI
 	isDiscovering     bool
 	awsConfig         *Client
+	csi               string
```
[the remark this answers is not in the document]
Comment (oxisto) [reply service/discovery/aws/compute.go:50]: Yes it it.

**Review by anatheka — APPROVED** [review]
LGTM

**Pull request merged**